packages/next-frontend/src/components/create-campaign/index.tsx

```diff
@@ -13,18 +17,34 @@ enum View {
 }
 
 export function CreateCampaign() {
+    const t = useTranslations("new_campaign");
     const [payload, setPayload] = useState<CampaignPayload>({});
     const [view, setView] = useState<View>(View.form);
+    const chainId = useChainId();
 
-    function handlePayloadOnChange(part: CampaignPayloadPart) {
-        setPayload({ ...payload, ...part });
-    }
+    // TODO: add complete validation
+    const malformedPayload = useMemo(() => {
+        return (
+            !payload.amm ||
+            !payload.pool ||
+            !payload.startDate ||
+            !payload.endDate
+        );
+    }, [payload]);
+
+    useEffect(() => {
+        setPayload({});
+    }, [chainId]);
 
     // TODO: remove
     useEffect(() => {
         console.log(JSON.stringify(payload, null, 4));
     }, [payload]);
 
+    function handlePayloadOnChange(part: CampaignPayloadPart) {
```

Comment:
useCallback?

Reply:
Fixed in 64b94f25b91cbb89a3677a6ed45bb3df5ea96415.